### ISSUE TITLE
Create PESTO option classes

### ISSUE THREAD

**Issue by dweindl** [ICB-DCM/PESTO#6]
Pass PESTO options as class instances.

Therefore, create classes:
* PestoOptions (for getMultiStarts getParameterProfiles getPropertyProfiles)
* <del>PestoParameter (for getMultiStarts getParameterProfiles getPropertySamples)</del>
* PestoPlottingOptions (for plot*.m)

Should look something like this:

PestoOptions
-----------------
````
   .obj_type ... type of objective function provided
       = 'log-posterior' (default) ... algorithm assumes that
               log-posterior or log-likelihood are provided and perfroms
               a maximization of the objective function.
       = 'negative log-posterior' ... algorithm assumes that negative
               log-posterior or negative log-likelihood are provided and
               perfroms a minimization of the objective function.
   .comp_type ... type of computations
       = 'sequential' (default) ... classical sequential (in core) method
       = 'parallel' ... multi-core method exploiting parfor
   .fmincon ... options for fmincon (the local optimizer)
   .n_starts ... number of local optimizations (default = 20).
   .init_threshold ... log-likelihood / log-posterior threshold for 
       initialization of optimization (default = -inf).
   .proposal ... method used to propose starting points
       = 'latin hypercube' (default) ... latin hypercube sampling
       = 'uniform' ... uniform random sampling
       = 'user-supplied' ... user supplied function parameters.init_fun
   .rng ... initialization of random number generator (default = 0).
       = any ral number r => random generator is initialized with r.
       = [] ... random number generator is not initialized.
       (Initializing the random number generator with a specific seed can be
       helpful to reproduce problems.)
   .mode ... output of algorithm
       = 'visual' (default) ... plots are gnerated which show the progress
       = 'text' ... optimization results for multi-start is printed on screen
       = 'silent' ... no output during the multi-start local optimization
   .fh ... handle of figure in which results are printed. If no
       handle is provided, a new figure is used.
   .plot_options ... plot options for plotMultiStarts.m.
   .save ... determine whether results are directly saved
       = false (default) ... results are not saved
       = true ... results are stored to an extra folder
   .trace ... determine whether objective function, parameter values and
   computation time are stored over iterations
       = false (default) ...  not saved
       = true ... stored in fields par_trace, fval_trace and time_trace
   .tempsave ... determine whether intermediate results are stored every
   10 iterations
       = false (default) ...  not saved
       = true ... results are stored to an extra folder
   .foldername ... name of the folder in which results are stored.
       If no folder is provided, a random foldername is generated.
   .start_index ... vector of indexes which starts should be performed.
       default is 1:n_starts
   .resetobjective ... clears the objective function before every
       multi-start.
       = false ... (default) persistent variables are preserved.
       = true ... remove all temporary/persistent variables.
       WHEN TRUE THIS OPTION REMOVES ALL OBJECTIVE FUNCTION BREAK POINTS
   .optimizer ... specifies which optimizer should be used
       = 'fmincon' ... (default) fmincon
       = 'minibatch' ... uses a minibatch optimization approach
   .optim_options ... struct with options for minibatch optimization
       .isMinibatch ... logical: perform full batch or minibatch optim
           = false ... deterministic optimization
           = true ... minibatch optim, Obj function must be adapted
       .nDatasets ... number of measurement points, only if isMinibatch
       .nBatchdata ... Size of Minibatches, only if isMinibatch
       .nOptimSteps ... number of maximum optimization steps
       .model ... String with the model name for AMICI, may be left void
       .method ... optimization method, to be chosen from
           = 'standard' ... stochastic gradient descent, standard method
           = 'momentum' ... sgd with momentum
           = 'nesterov' ... sgd with Nesterov momentum function
           = 'rmsprop' ... adaptive step size for each parameter
           = 'rmspropnesterov' ... with additional momentum
           = 'adam' ... adaptive method
           = 'adadelta' ... adaptive method
       .hyperparams ... struct containing the hyperparameters 
           (e.g. learning rate) for the opt-method, must fit with chosen 
           method (see documentation there)</pre>
````
<del>
PestoParameter
---------------------
````
    .number ... number of parameter
    .guess ... initial guess of parameter
    .min ... lower bound for parameter values
    .max ... upper bound for parameter values
    .name = {'name1',...} ... names of the parameters
    .init_fun ... function to draw starting points for local optimization
````
</del>

PestoPlottingOptions
--------------------------
````
   .hold_on ... indicates whether plots are redrawn or whether something
       is added to the plot
       = 'false' (default) ... new plot
       = 'true' ... extension of plot
   .interval ... selection mechanism for x limits
       = 'dynamic' (default) ... x limits depending on analysis results
       = 'static' ... x limits depending on parameters.min and .max or on
          user-defined bound options.bounds.min and .max. The later are
          used if provided.
   .bounds ... bounds used for visualization if options.interval = 'static'
       .min ... lower bound
       .max ... upper bound
   .P ... options for profile plots
       .plot_type ... plot type
           = 0 (default) ... no plot
           = 1 ... likelihood ratio
           = 2 ... negative log-likelihood
       .col ... color of profile lines (default: [1,0,0])
       .lw ... line width of profile lines (default: 1.5)
   .S ... options for sample plots
       .plot_type ... plot type
           = 0 (default if no samples are provided) ... no plot
           = 1 (default if samples are provided) ... histogram
       .col ... color of histogram (default: [0.7,0.7,0.7])
       .bins ... number of histogram bins (default: 30)
           = 'optimal' ... selection using Scott's rule
           = 'conservative' ... selection using Scott's rule / 2
           = N (with N being an integer) ... N bins
   .MS ... options for multi-start optimization plots
       .plot_type ... plot type
           = 0 (default if no MS are provided) ... no plot
           = 1 (default if MS are provided) ... likelihood ratio and
               position of optima above threshold
           = 2 ... negative log-likelihood and position of optima 
               above threshold
       .col ... color of local optima (default: [1,0,0])
       .lw ... line width of local optima (default: 1.5)
   .A ... options for distribution approximation plots
       .plot_type ... plot type
           = 0 (default if no MS are provided) ... no plot
           = 1 (default if MS are provided) ... likelihood ratio
           = 2 ... negative log-likelihood
       .col ... color of approximation lines (default: [0,0,1])
       .lw ... line width of approximation lines (default: 1.5)
   .boundary ... options for boundary visualization
       .mark ... marking of profile points which are on the boundary
           = 0 ... no visualization
           = 1 (default) ... indicates points which ar close to the
               boundaries in one or more dimensions.
       .eps ... minimal distance from boundary for which points are
               consider to e close do the boundary (default = 1e-4). Note
               that a one-norm is used.
   .CL ... options for confidence level plots
       .plot_type ... plot type
           = 0 (default) ... no plot
           = 1 ... likelihood ratio
           = 2 ... negative log-likelihood
       .alpha ... visualized confidence level (default = 0.95)
       .type ... type of confidence interval
           = 'point-wise' (default) ... point-wise confidence interval
           = 'simultanous' ... point-wise confidence interval
           = {'point-wise','simultanous'} ... both
       .col ... color of profile lines (default: [0,0,0])
       .lw ... line width of profile lines (default: 1.5)
   .op2D ... options used for 2D plot to position subplot axes.
       .b1 ... offset from left and bottom border (default = 0.15)
       .b2 ... offset from left and bottom border (default = 0.02)
       .r ... relative width of subplots (default = 0.95)
   .add_points ... option used to add additional points, e.g. true
           parameter in the case of test examples
       .par ... n x m matrix of m additional points
       .col ... color used for additional points (default = [0,0.8,0]).
                  This can also be a m x 3 matrix of colors.
       .ls ... line style (default = '--').
       .lw ... line width (default = 2).
       .m ... marker style (default = 'd').
       .ms ... line width (default = 8).
   .legend ... legend options
       .color ... background color (default = 'none').
       .box ... legend outine (default = 'on').
       .orientation ... orientation of list (default = 'vertical')
````

**Comment by dweindl** [comment]
Started work on branch feature_option_classes 3968d65

**Comment by dweindl** [comment]
Merged in 24c4f1a9d275eff63b40c26cf08b21c8f57fefe6
